### PR TITLE
#542: support secrets from variables in settings.xml

### PR DESCRIPTION
**Pull request by hohwille**
Implements issue #542

### CODE REVIEW

**Comment by hohwille** [comment]
Should we change our `settings.xml` template now to be ready for production-line or should we leave this up to the projects setting up their own settings git (to avoid annoying users who just want to jumpstart the IDE)?

**Comment by hohwille** [comment]
Shellcheck broke the build. Sorry, I should have checked in advance. `SC2155` is already fixed (trivial) but suggested pattern to fix `SC2013` does not work.
I changed this line
```
          for var in $( grep -o '\$\[.*\]' "${SETTINGS_PATH}/devon/conf/.m2/settings.xml" | sort -u)
```
to this line:
```
          grep -o '\$\[.*\]' "${SETTINGS_PATH}/devon/conf/.m2/settings.xml" | sort -u | while IFS= read -r var
```
As suggested here:
https://github.com/koalaman/shellcheck/wiki/SC2013

But testing reveals that then the entire thing is not working anymore and the variable `var` is set to the empty value.
I will rather add an exclusion of this shellcheck warning.
If some expert like @markusschuh may have a hinter later how to properly solve this `SC2013` issue, we can sill go for it but I wont let this issue get blocked by some shellcheck suggestion that does not seem to work.

File: documentation/conf.asciidoc
Comment (maybeec) [inline documentation/conf.asciidoc:26]: Could you provide a link here to the section which explains what batch mode is?

File: scripts/src/main/resources/scripts/command/mvn
Comment (maybeec) [inline scripts/src/main/resources/scripts/command/mvn:45]: Does this allow empty values? What if you don't have passwords at the moment? E.g. A person did not setup sonatype login yet?

**Review by maybeec — CHANGES_REQUESTED** [review]

**Comment by maybeec** [comment]
> Should we change our `settings.xml` template now to be ready for production-line or should we leave this up to the projects setting up their own settings git (to avoid annoying users who just want to jumpstart the IDE)?
> 
> https://github.com/devonfw/ide-settings/blob/9503a2dd6945a0cb38eac1ee5e775032b65ede60/devon/conf/.m2/settings.xml#L23

A valid point. Another suggestion would be to allow configuring mandatory credentials and optional ones by naming conventions on the variable names in the templates. Thus people get prompted on installation with only the mandatory set and we provide devon ide commands to reset the mandatory passwords (in case of password change) and one command to set all 'not yet set credentials.
Wdyt?

File: scripts/src/main/resources/scripts/command/mvn
Comment (hohwille) [reply scripts/src/main/resources/scripts/command/mvn:45]: sure. You can type any password, login, or nonsense here including the empty string. I tested with empty string and `mvn -ep` will give you `{}` that then ends up as the replacement. I think this is OK.

File: documentation/conf.asciidoc
Comment (hohwille) [reply documentation/conf.asciidoc]: OK, "batch mode" was not documented in asciidoc before either. Added it and linked here.

**Comment by hohwille** [comment]
> A valid point. Another suggestion would be to allow configuring mandatory credentials and optional ones by naming conventions on the variable names in the templates. Thus people get prompted on installation with only the mandatory set and we provide devon ide commands to reset the mandatory passwords (in case of password change) and one command to set all 'not yet set credentials.
Wdyt?

mhm, I was rather thinking that when you do not provide a settings URL on setup (by just hitting return) then the variables are not resolved (like in batch mode). I do not want to overengineer this. If your password has changed or your recieve a password after the setup you can simply delete your `settings.xml` file and recreate it via `devon mvn setup` that will prompt again for the variables/passwords.

**Review by maybeec — APPROVED** [review]

**Pull request merged**
